docs/source/cost.rst

```diff
@@ -0,0 +1,329 @@
+        fill='forward',
+        timezone='UTC'
+    )
+
+errors in the timeseries before 2020-05-01T12:00 are not included in
+the final calculation.
+
+
+Error Band Cost
+===============
+
+The error band cost model is defined using :py:class:`.ErrorBandCost`
+and implemented for deterministic forecasts by
+:py:func:`solarforecastarbiter.metrics.deterministic.error_band_cost`.
+Each of `bands` is a :py:class:`.CostBand` that describes the range of
+errors the band applies to and the parameters for one of the costs
+above. For example,
+
+.. code-block:: python
+
+    import datetime
+    from solarforecastarbiter import datamodel
+
+    cost_model = datamodel.ErrorBandCost(
+        bands=[
+            datamodel.CostBand(
+                error_range=(-5.0, 20.5),
+                cost_function='constant',
+                cost_function_parameters=datamodel.ConstantCost(
+                    cost=33.0,
+                    net=True,
+                    aggregation='sum'
+                )
+            ),
+            datamodel.CostBand(
+                error_range=(20.5, float('inf')),
+                cost_function='timeofday'
+                cost_function_parameters=datamodel.TimeOfDayCost(
+                    cost=[3.3, 1.2],
+                    times=[datetime.time(hour=15), datetime.time(hour=20)],
+                    net=True,
+                    aggregation='sum',
+                    fill='forward'
+                )
+            )
+        ]
+    )
+
+defines a cost that will apply a constant cost of $33.0 / unit error
+for all errors in the range [-5.0, 20.5]. For errors > 20.5, the time
+of day cost applies. The errors within each band are aggregated
+according to the `aggregation` and `net` parameter of the band
+parameters, but the total cost is the sum of all error bands.
+
+Band error ranges are evaluated in order and any errors outside the
+list of ranges *are not evaluated*. Thus, for the model described by
+
+.. code-block:: python
+
+    from solarforecastarbiter import datamodel
+
+    cost_model = datamodel.ErrorBandCost(
+        bands=[
+            datamodel.CostBand(
+                error_range=(-5.0, 5.0),
+                cost_function='constant',
+                cost_function_parameters=datamodel.ConstantCost(
+                    cost=2.0,
+                    net=True,
+                    aggregation='mean'
+                )
+            ),
+            datamodel.CostBand(
+                error_range=(-10.0, 10.0),
+                cost_function='constant',
+                cost_function_parameters=datamodel.ConstantCost(
+                    cost=4.0,
+                    net=True,
+                    aggregation='sum'
+                )
+            )
+        ]
+    )
+
+errors in the range [-5, 5] have cost of $2.0 / unit error. Errors
+that are outside [-5, 5] but within [-10, 10], that is errors in the
+range [-10, 5) or (5, 10] have a cost of $4.0 / unit error. Errors
```

Comment:
would the exact same behavior happen with the following?

```python
        bands=[
            datamodel.CostBand(
                error_range=(-5.0, 5.0),
                cost_function='constant',
                cost_function_parameters=datamodel.ConstantCost(
                    cost=2.0,
                    net=True,
                    aggregation='mean'
                )
            ),
            datamodel.CostBand(
                error_range=(-10.0, -5.0),
                cost_function='constant',
                cost_function_parameters=datamodel.ConstantCost(
                    cost=4.0,
                    net=True,
                    aggregation='sum'
                )
            ),
            datamodel.CostBand(
                error_range=(5.0, 10.0),
                cost_function='constant',
                cost_function_parameters=datamodel.ConstantCost(
                    cost=4.0,
                    net=True,
                    aggregation='sum'
                )
            )
        ]
```

Reply:
If yes, it might be good to explicitly state that (and test that - I haven't looked at them yet). It's certainly more convenient to specify fewer bands to accomplish the same thing.

Reply:
Yes, that is certainly the idea.